### ISSUE TITLE
Allow tag aliases to be reversed with a rename

### ISSUE THREAD

**Issue by nottalulah** [danbooru/danbooru#5364]
```
create alias tag1 -> tag2
```

If you want to reverse the alias, `create alias tag2 -> tag1` will work. Using a rename instead will cause the BUR to fail with `ActiveRecord::RecordInvalid: Cannot alias or implicate a tag to itself, tag1 is already aliased to tag2`.

**Comment by nonamethanks** [comment]
I don't think this should be allowed, I would say this is unexpected/undefined behavior. 

The BUR validation should reject this kind of operation instead of failing on approval. Someone trying to reverse and remove an alias should do it in multiple steps, first removing the alias and then renaming the tag.

**Comment by nonamethanks** [comment]
Addressed by 1f50db722dd612c6095b2b2abfbc305d3565642e
